mesa/datacollection.py

```diff
@@ -222,7 +222,9 @@ def get_model_vars_dataframe(self):
                 "No model reporters have been defined in the DataCollector, returning empty DataFrame."
             )
 
-        return pd.DataFrame(self.model_vars)
+        df = pd.DataFrame.from_dict(self.model_vars)
+        df.index.name = "Step"
+        return df
```

Comment:
Have you tried this with batchrunner? If it is single run the step is implicitly the tick (see line 216) but if you make the step explicitly the tick then I think you will start overwriting results when doing parameter sweeps.

Reply:
Apologies for the multiple messages had some extra time free up so was able to explore further.

Please disregard, the above and to be explicit, `batchrunner` has its own `get_model_vars_dataframe` but is deprecated and `batch_run` returns a list of dictionary results that is easily converted.

Two comments: 
1 - This change would explicitly show that the index is the "Step"? I don't have any issues with this change as it is consistent with principle of least surprise and make is constant with agent dataframe.  But could you also please update docstrings as this would make it no longer implicit. 

2. `pd.DataFrame(self.model_vars)` and `pd.DataFrame.from_dict(self.model_vars)` is there a particular reason for changing it?